src/db/collections/outputs/mod.rs

```diff
@@ -615,6 +750,75 @@ mod analytics {
             .unwrap_or_default())
         }
 
+        /// Gathers byte cost and storage deposit analytics and updates the analytics from the previous ledger index.
+        ///
+        /// NOTE: The `prev_analytics` must be from `ledger_index - 1` or the results are invalid.
+        #[tracing::instrument(skip(self), err, level = "trace")]
+        pub async fn update_ledger_size_analytics(
+            &self,
+            prev_analytics: &mut LedgerSizeAnalytics,
+            ledger_index: MilestoneIndex,
+        ) -> Result<(), Error> {
+            let (created, consumed) = tokio::try_join!(
+                async {
+                    Result::<_, Error>::Ok(self.aggregate::<LedgerSizeAnalytics>(
+                        vec![
+                            doc! { "$match": {
+                                "metadata.booked.milestone_index": ledger_index,
+                                "metadata.spent_metadata.spent.milestone_index": { "$ne": ledger_index }
+                            } },
+                            doc! { "$group" : {
```

Comment:
I really think we should get rid of the redundancy between the pipeline stages. There is high risk for bugs if the two queries (accidentally) diverge.

Reply:
Not in this PR though, right?